src/components/Form/From/NumericInput/helpers.ts

```diff
@@ -40,15 +40,20 @@ export function handleOnKeyPressNumericInput(e: KeyboardEvent): void {
 
 export function handleOnPasteNumericInput(e: ClipboardEvent, maxDecimals: number): string {
   const inputElement = e.target as HTMLInputElement;
-  const targetValue = inputElement.value;
+  const { value, selectionStart, selectionEnd } = inputElement;
 
   const clipboardData = sanitizeNumericInput(e.clipboardData?.getData('text/plain') || '');
-  const combinedValue = targetValue + clipboardData;
+
+  const combinedValue = value.slice(0, selectionStart || 0) + clipboardData + value.slice(selectionEnd || 0);
 
   const [integerPart, ...decimalParts] = combinedValue.split('.');
   const sanitizedValue = integerPart + (decimalParts.length > 0 ? '.' + decimalParts.join('') : '');
 
   e.preventDefault();
   inputElement.value = trimToMaxDecimals(sanitizedValue, maxDecimals);
+
+  const newCursorPosition = (selectionStart || 0) + clipboardData.length;
```

Comment:
This is already perfect. If you want it to be more than perfect, then you could change this line to 

```suggestion
  const newCursorPosition = (selectionStart || 0) + clipboardData.length - (combinedValue.length - sanitizedValue.length);
```

Reason: the cursor it not at the right position if pasting a valid number with a `.` if there is already a `.` in the number.

Reply:
@TorstenStueber Thanks for noticing it, I'll improve it. This input is so battle-tested and has so many test-cases that we can create a npm package named `decimal-input`